aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -82,6 +84,34 @@ impl TransactionMetadata {
                 TransactionPayload::Multisig(m) => Some(m.clone()),
                 _ => None,
             },
+            raw_txn_hash: match txn.authenticator_ref() {
+                MultiAgent {
+                    sender,
+                    secondary_signer_addresses,
+                    secondary_signers,
+                } => {
+                    let raw_txn = RawTransactionWithData::new_multi_agent(
+                        txn.clone().into_raw_transaction(),
+                        secondary_signer_addresses.clone(),
+                    );
+                    raw_txn.hash().to_vec()
+                },
+                FeePayer {
+                    sender,
+                    secondary_signer_addresses,
+                    secondary_signers,
+                    fee_payer_address,
+                    fee_payer_signer,
+                } => {
+                    let raw_txn = RawTransactionWithData::new_fee_payer(
+                        txn.clone().into_raw_transaction(),
+                        secondary_signer_addresses.clone(),
+                        *fee_payer_address,
+                    );
+                    raw_txn.hash().to_vec()
+                },
+                _ => txn.raw_transaction_ref().hash().to_vec(),
```

Comment:
This is really annoying so I hope we can do it after the refactoring... and force people to use the new format.

Reply:
Changing it to a new format will certainly impact many parts of the existing systems, though we're not exactly sure to what extent. It's a significant change that will likely take a lot of time, and we need to be careful to ensure backward compatibility.